### PR TITLE
Java Dates

### PR DESCRIPTION
**Pull request by rastin-sh**
wsg
honestly i don't write stuff and i suck at it this prob gonna look like a 12yo work but yeah. honestly i'm not too happy with this feel like it can get alot better but ya gotta start someplace and i'd rather figure out how you guys actually prefer to do stuff before adding alot more content. 

im absolutely ready to get crushed by a code review lol

hope you guys like it tho

### CODE REVIEW

**Comment by bowbahdoe** [comment]
@rastin-sh Its a start. I will read it more closely ronight. My immediate feedback is that I don't want to even tell people about `java.util.Date` until it becomes needed for something (like JDBC.)

We are actually about ready to cover this as a topic though.

File: src/dates/date.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+As a beginner one of the most interesting topics you’ll learn are Dates. Even though it might sound a bit boring, it's one of the key things you might know; from your Database, API, GUI, and more learning how to deal with dates properly is a key factor in writing good applications. 
+
+# Before we start
+An important detail you must remember is Java was rushed for the web, thus you have a lot of quirky and somewhat stupid ways to do stuff. The goal of this guide will be to teach you how to use the latest Java APIs to write efficient good code. I’ll aim to teach you how not to code anything hard and rather use standardized ISO for a scalable application.
```
Comment (bowbahdoe) [inline src/dates/date.md:4]: I am not even gonna disagree with this characterization, but I would rather talk about this stuff in the context of a "full history." or in the context of telling folks how social factors affect code development

File: src/dates/date.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
+As a beginner one of the most interesting topics you’ll learn are Dates. Even though it might sound a bit boring, it's one of the key things you might know; from your Database, API, GUI, and more learning how to deal with dates properly is a key factor in writing good applications. 
+
+# Before we start
+An important detail you must remember is Java was rushed for the web, thus you have a lot of quirky and somewhat stupid ways to do stuff. The goal of this guide will be to teach you how to use the latest Java APIs to write efficient good code. I’ll aim to teach you how not to code anything hard and rather use standardized ISO for a scalable application. 
+
+# How to work with Dates
+## The Date API 
+
+As of `Java 8 SE` with the release of the [Date Class](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html) defined as:
```
Comment (bowbahdoe) [inline src/dates/date.md:9]: Sentences like these date the content. Keep in mind that we are writing assuming anonymous main methods are already in the language - which hasn't happened yet.

Someone learning Java 25 might be interested when stuff was added, but I don't feel the need to call attention to it. Also Java 8 wasn't when Date was added. That was Java 1.

File: src/dates/date.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+An important specification to take note of is:
+> This class does not store or represent a time or time zone. Instead, it is a description of the date, as used for birthdays. It cannot represent an instant on the timeline without additional information such as an offset or time zone.
+## Difference & Which one to use 
+#### Issues with the Date/Time API’s 
+Further reading the [LocalDate](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html) class will lead you to “This class is immutable and thread-safe.” This brings us to our first problem.
+- Both the [Date](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html) and [Calender](https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html) were not thread-safe. If you're a bit confused as to what it means - Threads are a way to implement concurrency with a computer application this allows for multiple tasks to be running concurrently rather than sequentially (If you have a programming background you can see this as `async`).
```
Comment (bowbahdoe) [inline src/dates/date.md:28]: We haven't talked about threads yet, so I don't assume they are ready to talk about thread safety/unsafety.

**Comment by bowbahdoe** [comment]
@rastin-sh I left some comments but stopped relatively early since they all have a theme.

Something I should make clear in the contribution docs + maybe other places is that what I really want to maintain is a "topological sort" of topics. I.E. I don't ever want to introduce or talk about something that someone reading in order wouldn't have enough info to grasp.

All the talk about "poor design," the history of Java, thread safety, ISO times, etc. is lacking the necessary background info in the book as written so far.

Its also not in the format as the rest of the book. So far we've introduced larger things like this in sections with relatively bite-sized pages. This is a chunker.

But I will be writing about time types soon and I will use this as a reference - I'll be able to give better feedback at that point.

File: src/dates/date.md (hunk 2 above)
Comment (rastin-sh) [reply src/dates/date.md:9]: wasn't it released with jdk 1(more specifically jdk 1.1) with java 8? I'm tryna search online and find something that implies it was released with java 1 but I don't see any, also oracles java 1 core libraries list won't load up for some reason

**Comment by rastin-sh** [comment]
@bowbahdoe sure I get it. also one thing.
> Its also not in the format as the rest of the book. So far we've introduced larger things like this in sections with relatively bite-sized pages. This is a chunker.

I honestly don't think you could necessarily make this smaller without changing the format but as I mentioned above I'm not really good with writing so I could see your perspective as this is quite a mess of a "article". I guess in the future you could possibly reference the other parts such as `DateTimeFormatter`, `Locale`, and others which aren't necessarily closely tied with Dates as a whole. Not sure what to do with this pole at this moment, would u like me to close it?

**Comment by bowbahdoe** [comment]
Leave the PR open for now. It's useful for me as a reminder and I will try to restructure the info I can

**Comment by bowbahdoe** [comment]
@rastin-sh thank you for your patience btw. Circling around this topic like a bird right now.

**Comment by r0sieeee** [comment]
> @rastin-sh thank you for your patience btw. Circling around this topic like a bird right now.

well ive been expecting something back. also never knew you could be a bird. cool.

btw this is one of my many alts i was too lazy to get back on that other account.

**Comment by ryansamaeian** [comment]
hi mate, i did respond to your message as soon as i read it. now after a month or so i think i come back to find out my account is now disabled. now i could rant about it all day but id rather just follow up on this whole thing from this account if you dont mind cause i honestly am interested in knowing more about dates in java

**Comment by bowbahdoe** [comment]
@k1nxx I pushed a section on times. I'll close this for now, but I think I got all the info I wanted from your changes. I'll look back here if/when i come around to a "Time 2" section

**Pull request closed without merging**